File: MISA.AssetManagement.Api/MISA.AssetManagement.Fresher.Api/MISA.AssetManagement.Core/Entities/FixedAssetCategory.cs
using System;

namespace MISA.Core.Entities
{
    /// <summary>
    /// Thông tin loại tài sản cố định
    /// CreatedBy: TTVinh (14/11/2025)
    /// </summary>
    public class FixedAssetCategory
    {
        /// <summary>
        /// ID của loại tài sản cố định
        /// </summary>
        public Guid FixedAssetCategoryId { get; set; }

        /// <summary>
        /// Mã loại tài sản cố định
        /// </summary>
        public string FixedAssetCategoryCode { get; set; }

        /// <summary>
        /// Tên loại tài sản cố định
        /// </summary>
        public string FixedAssetCategoryName { get; set; }

        /// <summary>
        /// Tên viết tắt cho tài sản cố định
        /// </summary>
        public string? FixedAssetCategoryShortName { get; set; }

        /// <summary>
        /// Thời gian sử dụng (năm)
        /// </summary>
        public int LifeTime { get; set; }

        /// <summary>
        /// Tỷ lệ hao mòn (%)
        /// </summary>
        public decimal DepreciationRate { get; set; }

        /// <summary>
        /// Mô tả thêm về loại tài sản
        /// </summary>
        public string? Description { get; set; }

        /// <summary>
        /// Trạng thái hoạt động (true: đang hoạt động, false: ngừng sử dụng)
        /// </summary>
        public bool IsActive { get; set; }

        /// <summary>
        /// Ngày tạo bản ghi
        /// </summary>
        public DateTime? CreatedDate { get; set; }

        /// <summary>
        /// Người tạo bản ghi
        /// </summary>
        public string? CreatedBy { get; set; }

        /// <summary>
        /// Ngày chỉnh sửa gần nhất
        /// </summary>
        public DateTime? ModifiedDate { get; set; }

        /// <summary>
        /// Người chỉnh sửa gần nhất
        /// </summary>
        public string? ModifiedBy { get; set; }
    }
}

File: MISA.AssetManagement.Webapp/MISA.AssetManagement.Fresher.Webapp/src/domains/models/FixedAssetCategory.js
/**
 * @fileoverview FixedAssetCategory Model
 * @description Đại diện cho đối tượng "Loại tài sản" (Fixed Asset Category)
 * - Map trực tiếp với DTO từ backend.
 * - Cung cấp phương thức helper để hiển thị trong UI (filter, form...).
 * @createdBy TTVinh (18/11/2025)
 */

export class FixedAssetCategory {
  constructor(data = {}) {
    this.FixedAssetCategoryId = data.FixedAssetCategoryId || data.fixedAssetCategoryId || null
    this.FixedAssetCategoryCode = data.FixedAssetCategoryCode || data.fixedAssetCategoryCode || ''
    this.FixedAssetCategoryShortName = data.FixedAssetCategoryShortName || data.fixed_asset_category_short_name || data.fixedAssetCategoryShortName || ''
    this.FixedAssetCategoryName = data.FixedAssetCategoryName || data.fixedAssetCategoryName || ''
    this.LifeTime = data.LifeTime || data.lifeTime || 0
    this.DepreciationRate = data.DepreciationRate || data.depreciationRate || 0
  }

  get id() {
    return this.FixedAssetCategoryId
  }

  get code() {
    return this.FixedAssetCategoryCode
  }

  get shortName() {
    return this.FixedAssetCategoryShortName
  }

  get name() {
    return this.FixedAssetCategoryName
  }

  get lifeTime() {
    return this.LifeTime
  }

  get depreciationRate() {
    return this.DepreciationRate
  }

  toToolbarOption() {
    return {
      value: this.FixedAssetCategoryCode,
      label: this.FixedAssetCategoryName,
      fullName: this.FixedAssetCategoryName,
      id: this.FixedAssetCategoryId,
      lifeTime: this.LifeTime,
      depreciationRate: this.DepreciationRate
    }
  }

  toFormOption() {
    return {
      value: this.FixedAssetCategoryCode,
      label: this.FixedAssetCategoryShortName,
      fullName: this.FixedAssetCategoryName,
      id: this.FixedAssetCategoryId,
      lifeTime: this.LifeTime,
      depreciationRate: this.DepreciationRate
    }
  }

  toOption() {
    return this.toToolbarOption()
  }

  toSelectOption() {
    return this.toFormOption()
  }

  static fromApi(data) {
    return new FixedAssetCategory(data)
  }

  static fromApiArray(dataArray) {
    if (!Array.isArray(dataArray)) return []
    return dataArray.map(data => new FixedAssetCategory(data))
  }
}
